### PR TITLE
db: small molecule batch qc events

### PR DESCRIPTION
**Pull request by seanderickson**

### CODE REVIEW

File: django/db/views.py [hunk 1]
```diff
@@ -614,11 +615,53 @@ def smallMoleculeDetail(request, facility_salt_id):
             batches = SmallMoleculeBatch.objects.filter(smallmolecule=sm)
             if(len(batches)>1):
                 details['batchTable']=SmallMoleculeBatchTable(batches)
+                # 20150413 - proposed "QC Outcome" field on batch info removed per group discussion
```
Comment (jmuhlich) [inline django/db/views.py]: Delete this commented code.

File: django/db/views.py [hunk 2]
```diff
@@ -614,11 +615,53 @@ def smallMoleculeDetail(request, facility_salt_id):
...
+            details['facility_salt_batch'] = '%s-%s-%s' % (sm.facility_id,sm.salt_id,smb.facility_batch_id) 
+            # 20150413 - proposed "QC Outcome" field on batch info removed per group discussion
+            qcEvents = QCEvent.objects.filter(
+                facility_id_for=sm.facility_id,
+                salt_id_for=sm.salt_id,
+                batch_id_for=smb.facility_batch_id).order_by('-date')
+            fi = FieldInformation.manager.get_column_fieldinformation(
+                'qc_outcome', 'smallmoleculebatch')
+            if qcEvents:
+                # manually add in the qc_outcome field
```
Comment (jmuhlich) [inline django/db/views.py]: delete

File: django/db/views.py [hunk 3]
```diff
@@ -614,11 +615,53 @@ def smallMoleculeDetail(request, facility_salt_id):
...
+                #                 details['smallmolecule_batch']['qc_outcome'] = {
+                #                     'value': qcEvents[0].outcome,
+                #                     'fieldinformation': fi
+                #                     }
+                details['qcTable'] = QCEventTable(qcEvents)
+                #             else:
```
Comment (jmuhlich) [inline django/db/views.py]: delete!

File: django/db/views.py [hunk 4]
```diff
@@ -614,11 +615,53 @@ def smallMoleculeDetail(request, facility_salt_id):
...
         if smb:
             details['smallmolecule_batch']= get_detail(
                 smb,['smallmoleculebatch',''])
+            details['facility_salt_batch'] = '%s-%s-%s' % (sm.facility_id,sm.salt_id,smb.facility_batch_id) 
+            # 20150413 - proposed "QC Outcome" field on batch info removed per group discussion
+            qcEvents = QCEvent.objects.filter(
+                facility_id_for=sm.facility_id,
+                salt_id_for=sm.salt_id,
+                batch_id_for=smb.facility_batch_id).order_by('-date')
+            fi = FieldInformation.manager.get_column_fieldinformation(
```
Comment (jmuhlich) [inline django/db/views.py]: delete this fi=... line as discussed.

File: django/db/views.py [hunk 5]
```diff
@@ -3058,37 +3137,55 @@ def restricted_image(request, filepath):
...
+        _path = settings.STATIC_AUTHENTICATED_FILE_DIR
+        if(file_obj.relative_path):
+            _path = os.path.join(_path,file_obj.relative_path)
+        _path = os.path.join(_path,file_obj.filename)
         _file = file(_path)
-        logger.debug(str(('download_attached_file',_path,_file)))
+        logger.info(str(('download_attached_file',_path,_file)))
         wrapper = FileWrapper(_file)
         # use the same type for all files
         response = HttpResponse(wrapper, content_type='text/plain')
```
Comment (dpwrussell) [inline django/db/views.py]: Potentially application/octet-stream might be more appropriate if the type is unknown. Long term, ideally use the case specific file type if it is possible to determine

File: src/import_qc_events_batch.py [hunk 6]
```diff
@@ -0,0 +1,212 @@
+
```
Comment (dpwrussell) [inline src/import_qc_events_batch.py]: Python shebang here?

**Comment by seanderickson** [comment]
Note: will deploy this code to production after 5pm

**Pull request merged**